src/lib.rs

```diff
@@ -74,97 +74,119 @@ mod tests {
         // No need to handle our own vote
         // Participant 0 handles Parts
         // We already know this vote (it's ours), just checking that it gives IgnoringKnownVote
-        assert!(matches!(
-            dkg_state0.handle_signed_vote(part0.clone(), &mut rng),
-            Ok(VoteResponse::IgnoringKnownVote)
-        ));
-        assert!(matches!(
-            dkg_state0.handle_signed_vote(part1.clone(), &mut rng),
-            Ok(VoteResponse::WaitingForMoreVotes)
-        ));
-        let acks0 = assert_match!(
-           dkg_state0.handle_signed_vote(part2.clone(), &mut rng),
-           Ok(VoteResponse::BroadcastVote(acks)) => *acks
-        );
+        let res = dkg_state0
+            .handle_signed_vote(part0.clone(), &mut rng)
+            .unwrap()
+            .remove(0);
```

Comment:
Does indexing with `0` work here? That may clean up these tests a little.
```suggestion
        let res = dkg_state0.handle_signed_vote(part0.clone(), &mut rng)[0];
```